Continue work on unified release notes

da-woods opened the pull request:
This is largely just h-vetinari's changes fixing up the unified release notes, with a few of the remaining gaps filled in by me.

I can do any remaining changes that need to me made.

h-vetinari commented on CHANGES.rst:
```diff
@@ -1006,614 +1006,6 @@ Other changes
 
-vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
-vvv    TO DISTRIBUTE TO PROPER SECTIONS ABOVE    vvv
-vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
```
In an ideal world, those items being removed by 7245b4a0106d59dddaafa194a6b42f4eee83ffba still need to be either distributed to an appropriate section (or if not relevant anymore, deleted). Mainly those are items where I didn't know where to put them (e.g.: not clear between Python compat, pure python, Cython improvement, etc.). It's possible that we might want to add or change sections based on the rest of the lines here.

da-woods replied:
Ah thanks. I'll probably add an "optimisations" section and a "everything else" section.

h-vetinari commented:
Within the "Related Fixes" subsections, I generally tried to keep issue numbers ascending, even though those were obviously often not solved in the order that they were opened in. But from the POV of 3.0 vs. 0.29, that distinction doesn't really matter and so the other ordering was more obvious IMO. Items without an issue reference are broadly kept at the rough point where they occurred chronologically with the surrounding ones.

> I can do any remaining changes that need to me made.

Thanks a lot! Aside from good summaries (where we already have something pretty much everywhere), the one remaining point I had wanted to tackle in #5532 was what I mentioned in the above comment

da-woods commented:
My thought going through more of the list: I'm tempted to drop bug fixes to features. (I.e. the state as of Cython 3 is the main thing, and we can pretend we got everything right first time).

h-vetinari commented:
> My thought going through more of the list: I'm tempted to drop bug fixes to features. (I.e. the state as of Cython 3 is the main thing, and we can pretend we got everything right first time).

Fully agreed, that's the only thing where I dropped some items without double-checking, because the intermediate states released in the alphas (and their subsequent bugfixes) aren't relevant from the POV of 0.29 -> 3.0.

da-woods commented:
I think I've categorised everything and sorted everything (I ended up automating the sorting).

In the interests of getting these updates visible, I'm going to merge on Monday morning unless objections are made before then - we can always have minor fixups after that

h-vetinari commented on CHANGES.rst:
```diff
@@ -19,101 +19,158 @@ backported to 0.29.x and are not strictly speaking "new" in Cython 3.0.0.
+
+
+
+Compatibility with CPython and the Python C API
+-----------------------------------------------
+
+Since Cython 3.0.0 started development, CPython 3.8-3.11 were released.
+All these are supported in Cython, including experimental support for the
+in-development CPython 3.12. On the other end of the spectrum, support for
+Python 2.6 was dropped.
+
+Cython interacts very closely with the C-API of Python, which is where most
+of the adaptation work happens.
```
This section appears duplicated with the above. Haven't checked all items, but at least the title, intro and first few are the same.

h-vetinari reviewed (commented):
Aside from a duplicated section this looks very good IMO! I couldn't verify that the moved entries didn't cause further duplication (or dropping) of entries, but I'm certain it's an improvement over my first rough classification in any case.

da-woods commented:
Thanks for spotting that - it looks like I also removed the "Other Python implementations" section, most likely as part of the same error that led to the duplication.

> I couldn't verify that the moved entries didn't cause further duplication (or dropping) of entries

There some things I deliberately drop (e.g. auto-conversion to `cpdef enums` required about 3 fixups and I dropped some of those) so I suspect trying to verify that would be a bit futile.

h-vetinari commented on CHANGES.rst:
```diff
@@ -229,359 +216,1251 @@ Related fixes
 
-Interaction with numpy
-----------------------
+* Diverging from the usual behaviour, ``len(memoryview)``, ``len(char*)``
+and ``len(Py_UNICODE*)`` returned an unsigned ``size_t`` value.  They now
+return a signed ``Py_ssize_t``, like other usages of ``len()``.
```
```suggestion
* Diverging from the usual behaviour, ``len(memoryview)``, ``len(char*)``
  and ``len(Py_UNICODE*)`` returned an unsigned ``size_t`` value.  They now
  return a signed ``Py_ssize_t``, like other usages of ``len()``.
```

h-vetinari reviewed (commented):
I had a brief look at the [rendered](https://github.com/da-woods/cython/blob/71f0a0b3372528a607a10a926d41f4ca2f44fa72/CHANGES.rst) version, and one bullet point is broken.

da-woods commented:
As threatened I'm going to merge this, and any further fix-ups can proceed from there.

Thanks @h-vetinari
